Arm/State.lean

```diff
@@ -645,6 +690,13 @@ theorem read_mem_bytes_of_w :
     rw [n_ih]
   done
 
+theorem read_mem_bytes_w_of_read_mem_eq
```

Comment:
Add `state_simp_rules` tag.

Reply:
BTW, I know this ruleset is aggressive at the moment, but it's good to have all the relevant theorems in there in case we want to revert to doing symbolic simulation manually for whatever reason.

Reply:
Done